### ISSUE TITLE
Btrfs support

### ISSUE THREAD

**Issue by graywolf-at-work** [mcrute/tiny-ec2-bootstrap#12]
I've hacked together btrfs support for the root resize and I'm willing to take the time to upstream it. There are two points I need to mention:

1. It depends on cloud-utils

It is part of alpine-edge so I personally am fine with that, but not sure if you have an issue with it.

2. It obviously complicates code a bit since now resizing function needs to support both ext and btrfs

But it's not that bad :)

If both are ok with you, I'll prepare v pull request.

**Comment by tomalok** [comment]
I've taken a quick look at `cloud-utils` and it looks like several tools in there explicitly depend on `bash` and `python` -- so, those would end up being dependencies of `tiny-ec2-bootstrap` too, and make its installation much less tiny.

Perhaps the multi-FS resizing support portion could be ported or reverse engineered in a busybox-`ash`-friendly way and/or the inclusion (or exclusion) of other FS (and the necessary tools to manage them) could be made optional via build profile.

**Comment by mcrute** [comment]
I agree with @tomalok on the inclusion of `cloud-utils`, it's quite heavy in the dependencies it pulls in. We're trying to keep the Alpine images very lightweight as a base and allow users to layer on other things after the image is bootstrapped. Also, we build non-edge images so we can't take any dependencies on edge-only packages at this point.

I'm happy to take patches that add btrfs support but they'd have to do it in a way that does not pull in a bunch of additional dependencies (as judged by the total size of the dependency graph as well as the package size of all those dependencies) and does not use edge-only dependencies.

**Comment by dermotbradley** [comment]
Hi @mcrute 

> I agree with @tomalok on the inclusion of cloud-utils, it's quite heavy in the dependencies it pulls in. We're trying to keep the Alpine images very lightweight as a base and allow users to layer on other things after the image is bootstrapped. Also, we build non-edge images so we can't take any dependencies on edge-only packages at this point.

I took over Alpine maintainership of cloud-utils (along with cloud-init) a few months ago. I split cloud-init into several sub-packages specifically because of the heavy dependancies - there is now a cloud-init-growpart sub-package (which cloud-init uses) which has only 4 dependancies.

Also I have moved cloud-utils from testing to community so that when Alpine 3.13.0 is released later this year it will include cloud-utils.

BTW I believe you may have chatted to ncopa in the past about creating **official** Alpine AWS images. I talked to him also not long ago about official cloud images in general - I do think that Alpine is losing out by not having them unlike other distributions. I see tiny-ec2-bootstrap as complimentry to cloud-init on Alpine as you provide a lightweight solution specifically for AWS whereas cloud-init provides a more generic multi-cloud solution.
